src/Coordination/KeeperServer.cpp

```diff
@@ -111,7 +111,7 @@ void KeeperServer::startup()
     params.auto_forwarding_ = coordination_settings->auto_forwarding;
     params.auto_forwarding_req_timeout_ = coordination_settings->operation_timeout_ms.totalMilliseconds() * 2;
 
-    params.return_method_ = nuraft::raft_params::blocking;
+    params.return_method_ = nuraft::raft_params::async_handler;
```

Comment:
Main change. In this mode, NuRaft doesn't return the result from `append_entries`. But we haven't used it already for normal requests. Only session-id request relied on `append_entries` result. To avoid this I've added internal `ZooKeeperSessionIDRequest` and `ZooKeeperSessionIDResponse`. 
And now session-id requests processed like all other types of requests.